### PR TITLE
Add support for Azure, OpenAI, Palm, Anthropic, Cohere Models - using litellm

### PR DESCRIPTION
**Pull request by ishaan-jaff**
I'm the maintainer of litellm https://github.com/BerriAI/litellm - a simple & light package to call OpenAI, Azure, Cohere, Anthropic, Replicate API Endpoints

This PR adds support for models from all the above mentioned providers (by creating a class `liteLLMChatGPTBot`)

Here's a sample of how it's used:

```
from litellm import completion

## set ENV variables
# ENV variables can be set in .env file, too. Example in .env.example
os.environ["OPENAI_API_KEY"] = "openai key"
os.environ["COHERE_API_KEY"] = "cohere key"

messages = [{ "content": "Hello, how are you?","role": "user"}]

# openai call
response = completion(model="gpt-3.5-turbo", messages=messages)

# cohere call
response = completion("command-nightly", messages)

# anthropic call
response = completion(model="claude-instant-1", messages=messages)
```

### CODE REVIEW

**Comment by ishaan-jaff** [comment]
cc @iuiaoin can i get a review on this

File: bot/bot.py [hunk 1]
```diff
@@ -13,6 +15,13 @@ def __init__(self):
...
             self.bot = AzureChatGPTBot()
         else:
+            model = conf().get("model", "gpt-3.5-turbo")
+            if model in litellm.model_list:
+                # see litellm supported models here:
+                # https://litellm.readthedocs.io/en/latest/supported/
+                from bot.litellm import liteLLMChatGPTBot
+                self.bot = liteLLMChatGPTBot()
+
             from bot.chatgpt import ChatGPTBot
```
Comment (iuiaoin) [inline bot/bot.py:25]: The self.bot will always be assigned by ChatGPTBot instance, seems there needs to be  an else block here.

File: bot/bot.py [hunk 2]
```diff
@@ -13,6 +15,13 @@ def __init__(self):
...
 
             self.bot = ChatGPTBot()
```
Comment (iuiaoin) [inline bot/bot.py:27]: The self.bot will always be assigned by ChatGPTBot instance, seems there needs to be an else block here.

**Comment by iuiaoin** [comment]
@ishaan-jaff Thanks for your pull request! I got a comment for you

File: bot/bot.py (hunk 2 above)
Comment (ishaan-jaff) [reply bot/bot.py:24]: good catch updated this

**Comment by iuiaoin** [comment]
@ishaan-jaff LGMT，approved. Yet seems ci failed

**Comment by ishaan-jaff** [comment]
@iuiaoin fixed lint errors

**Pull request merged**